lightning_integration_test.py

```diff
@@ -6,25 +6,29 @@
 
 print("\nSTART Lightning integration tests\n")
 
-# Test echoing
+# Test echoing using httpie
 # Try a background fork/thread
 server_process = subprocess.Popen(['./lightning', 'simple_config'])
 
+# Spawn a shell process to act as hanging http request
+telnet_request_command = "telnet localhost 2020"
+telnet_request_proc = subprocess.Popen(telnet_request_command, stdout=subprocess.PIPE, shell=True)
+
 # TODO: Have intermediate logging throughout
 # TODO: Use Python unit test frameworks + logging libraries
 print('DEBUG: Lightning server started!')
 
-expected_response = b'GET / HTTP/1.1\r\nHost: localhost:8080\r\nUser-Agent: HTTPie/0.9.8\r\nAccept-Encoding: gzip, deflate\r\nAccept: */*\r\nConnection: keep-alive\r\n\r\n'
-actual_response = run(['http', 'localhost:8080'], stdout=subprocess.PIPE)
+expected_response = b'GET /echo HTTP/1.1\r\nHost: localhost:2020\r\nAccept-Encoding: gzip, deflate, compress\r\nAccept: */*\r\nUser-Agent: HTTPie/0.8.0\r\n\r\n'
+actual_response = run(['http', 'localhost:2020/echo'], stdout=subprocess.PIPE)
```

Comment:
Could we add a single short comment here to explain how this tests multithreading? For example, "The first request is hanging, waiting for input to telnet (which will never come), but since this echo request should be handled by a separate handler in a separate thread, it should still succeed and return a response instantly."